File: LifecycleExample/LifecycleExampleJava/app/src/main/java/com/tinlone/lifecycleexamplejava/MyApplication.java
package com.tinlone.lifecycleexamplejava;

import android.app.Application;

import androidx.lifecycle.ProcessLifecycleOwner;

public class MyApplication extends Application {

    private ProcessLifecycleObserver observer;

    @Override
    public void onCreate() {
        super.onCreate();

        observer = new ProcessLifecycleObserver();
        ProcessLifecycleOwner.get().getLifecycle().addObserver(observer);
    }

    public boolean isForeground() {
        return observer.isForeground();
    }
}

File: LifecycleExample/LifecycleExampleKotlin/app/src/main/java/com/tinlone/lifecycleexamplekotlin/MyApplication.kt
package com.tinlone.lifecycleexamplekotlin

import android.app.Application
import androidx.lifecycle.ProcessLifecycleOwner

class MyApplication() : Application() {

    private var observer: ProcessLifecycleObserver? = null

    override fun onCreate() {
        super.onCreate()
        observer = ProcessLifecycleObserver()
        ProcessLifecycleOwner.get().lifecycle.addObserver(observer!!)
    }

    fun isForeground(): Boolean {
        return observer!!.isForeground()
    }
}